ambry-frontend/src/main/java/com/github/ambry/frontend/AccountAndContainerInjector.java

```diff
@@ -84,6 +85,48 @@ public void injectAccountAndContainerForPostRequest(RestRequest restRequest, Res
     }
   }
 
+  /**
+   * Injects target {@link Account} and {@link Container} for PUT requests. This method also ensures required uri
+   * is present for the PUT requests that can account name/container name can parse from.
+   * @param restRequest The Put {@link RestRequest}.
+   * @param metricsGroup The {@link RestRequestMetricsGroup} to use to set up {@link ContainerMetrics}, or {@code null}
+   *                     if {@link ContainerMetrics} instantiation is not needed.
+   * @throws RestServiceException
+   */
+  public void injectAccountAndContainerForPutRequest(RestRequest restRequest, RestRequestMetricsGroup metricsGroup)
```

Comment:
For named blob put request, the source of truth is the blob url which looks like "/named/accountName/containerName/blobName", so client don't need to provide any accountName/containerName headers/blobId. Similar like post request, it will call router.putBlob first to get the blobId, and pass it to Id convertor. And the difference between the post request is post request will return blobId as a location header but in this put nameBlob case there's no need to return anything. So after discuss with @cgtz , we decided to generate the putBlobHandler to handler this case. And already named it as NamedBlobPutHandler. Feel free to let me know if you have any concern.

Reply:
Okay thanks for explaining.